### PR TITLE
[bitnami/kube-prometheus] operator/alertmanager serviceMonitor extraParams

### LINKED ISSUE #16086: [kube-prometheus / kube-state-metrics] Unable to scrape metrics with Istio injection enabled
### Name and Version

bitnami/kube-prometheus 8.5.0, bitnami/kube-state-metrics 3.3.5

### What is the problem this feature will solve?

Unable to scrape metrics with Istio injection enabled.

When having istio injection enabled at the namespace level (namespace where bitnami/kube-prometheus with bitnami/kube-state-metrics and bitnami/node-exporter, and bitnami/thanos are all installed), the pods from the following resources cannot be scraped:

- deploy/kube-prometheus-system-operator
- deploy/kube-prometheus-system-kube-state-metrics
- sts/alertmanager-kube-prometheus-system-alertmanager

This is due to the fact that the scraped endpoints must enforce a specific http scheme and tls_config, see [the istio doc here](https://istio.io/latest/docs/ops/integrations/prometheus/#tls-settings).

### What is the feature you are proposing to solve the problem?

Doing the same as for bitnami/thanos chart, i.e.

[kube-prometheus]
Adding the following fields:
- `operator.serviceMonitor.extraParameters` --> would control the` http` endpoint of the `ServiceMonitor` of the `operator`
- `alertmanager.serviceMonitor.extraParameters` --> would control the `http` endpoint of the `ServiceMonitor` of the `alertmanager`

[kube-state-metrics]
Adding the following field: `serviceMonitor.extraParameters` -->  would control both the `http` and `metrics` endpoints of the (only) `ServiceMonitor`

No need to enable this field for other components to fix this problem. For example: the `coredns` pods are in `kube-system` namespace therefore cannot have the istio sidecar injected and the issue will never appear, etc.

With these changes, we would be able to do the following for the `kube-state-metrics` pod for example, and have its metrics scraped (following the istio doc link I shared in the previous section):

```yaml
# values.yaml
serviceMonitor:
  enabled: true
  extraParameters:
    scheme: "https"
    tlsConfig:
      caFile: /etc/prom-certs/root-cert.pem
      certFile: /etc/prom-certs/cert-chain.pem
      keyFile: /etc/prom-certs/key.pem
      insecureSkipVerify: true
``` 

### What alternatives have you considered?

Patching the serviceMonitors with a controller as [patch-operator](https://github.com/redhat-cop/patch-operator) to keep GitOps pattern at best and avoid manual changes. Not really convenient at all.

### PR DESCRIPTION
**Pull request by leotomas837**
### Description of the change

Add extraParameters for kube-prometheus serviceMonitors (operator and alert manager).

### Benefits

Fix #16086

### Possible drawbacks

NA

### Applicable issues

- fixes #16086

### Additional information

NA

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/kube-prometheus/Chart.yaml [hunk 1]
```diff
@@ -35,4 +35,4 @@ sources:
   - https://github.com/bitnami/containers/tree/main/bitnami/prometheus
   - https://github.com/bitnami/containers/tree/main/bitnami/alertmanager
   - https://github.com/prometheus-operator/kube-prometheus
-version: 8.5.0
```
Comment (migruiz4) [inline bitnami/kube-prometheus/Chart.yaml:38]: As this PR adds new features for the chart, could you please bump a minor version `8.6.0`?

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi @leotomas837,

Thank you for your contribution! Could you please bump a minor version?

File: bitnami/kube-prometheus/Chart.yaml (hunk 1 above)
Comment (leotomas837) [reply bitnami/kube-prometheus/Chart.yaml:38]: Hi @migruiz4, just done

**Review by migruiz4 — APPROVED** [review]
Thank you for applying my suggestion, looks good to me!

**Pull request merged**